tests/unit/models/test_dflash_liger.py

```
@@ -0,0 +1,119 @@
+    def fail_if_called():
+        raise AssertionError("disabled Liger path must not load the optional extra")
+
+    monkeypatch.setattr("scripts.train.load_liger_dflash_kernels", fail_if_called)
+
+    assert _resolve_dflash_kernels(argparse.Namespace(use_liger_kernel=False)) is None
+
+
+def test_missing_liger_extra_has_actionable_error(monkeypatch):
+    original_import = builtins.__import__
+
+    def missing_liger(name, globals=None, locals=None, fromlist=(), level=0):
```

Ruff (0.15.21)

[error] Function argument `globals` is shadowing a Python builtin

(A002)